exonum-java-binding/core/src/main/java/com/exonum/binding/core/runtime/TransactionInvoker.java

```diff
@@ -54,9 +56,14 @@ void invokeTransaction(int transactionId, byte[] arguments, TransactionContext c
       throws TransactionExecutionException {
     checkArgument(transactionMethods.containsKey(transactionId),
         "No method with transaction id (%s)", transactionId);
+    TransactionMethodObject transactionMethodObject = transactionMethods.get(transactionId);
+    Optional<Serializer> argumentsSerializer = transactionMethodObject.getArgumentsSerializer();
```

Comment:
TransactionMethodObject just exposes all of its properties. I suggest moving
  the serialization logic into that class to separate responsibilities (routing vs
  parameter resolution, mapping and invocation).